libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -296,7 +295,15 @@ class NavigationRoute internal constructor(
      */
     val upcomingRoadObjects = nativeRoute.routeInfo.alerts.toUpcomingRoadObjects()
 
-    internal val waypoints: List<Waypoint> by lazy { nativeRoute.waypoints.mapToSdk() }
+    val waypoints: List<DirectionsWaypoint>? by lazy {
```

Comment:
I wonder if this should be added to `NavigationRoute#equals`. @LukasPaczos 
I suppose that directionsRoute equality is equivalent to directionsRoute + waypoints (except for metadata) equality. Because I don't imagine 2 routes with different waypoints and the same everything else (legs, steps, geometries, etc.).
But should the metadata affect the equality? It contains dynamic values like charge_at_arrival and so on. It's important in terms of EV station charging experience.
@LukasPaczos

Reply:
Yes, we should fully include any relevant information of the route tracked by `NavigationRoute` in the equals check. This covers waypoints and their metadata. The goal of this customized equals is avoiding comparing other routes from the original response beyond the one that's tracked by this `NavigationRoute` instance.

Reply:
That makes sense to me. The request parameters were different, so the routes are different in our view as well.